draft-ietf-ohai-ohttp.md

```diff
@@ -976,7 +977,11 @@ removing unknown fields removes this privacy risk.
 Secondly, generic implementations are often configured to augment requests with
 information about the client, such as the Via field or the Forwarded field
 {{?FORWARDED=RFC7239}}.  A proxy MUST NOT add information about the client
-identity when forwarding requests.
+identity when forwarding requests without agreement from the client.  In
+particular, a proxy can attach additional information about a client to each
+request, provided this behavior is known to the client.  Clients need to be
+aware any additional bit of information added by the proxy can reduce the size
+of the client anonymity set behind the proxy by half.
```

Comment:
```suggestion

A proxy MAY add information to requests if the client is aware of the nature of
the information that could be added.  The client does not need to be aware of
the exact value added for each request, but needs to know the range of possible
values the proxy might use.  Clients need to be aware any information added by
the proxy can reduce the size of the anonymity set of clients at a server.
```

Aside from the extra text, I wanted to add a small note about the last change I'm proposing.

Though "one bit of information" has a very specific information theoretic meaning that distinctly means that the anonymity set is halved, that specific meaning is likely lost on people reading this.  A reader might easily conclude that it is OK to add a one-bit field on the basis that users won't suffer anything worse than a halving of the anonymity set.  However, the extent to which that is true depends on how that bit is allocated.

To take an extreme example, imagine a one bit field called "Is-Chris-Wood" that is set to 0 for everyone except @chris-wood.  In that case - despite the net information gain for the server being very small - the net information loss for Chris is massive.  That's an outcome that is clearly unacceptable.

I almost want to suggest that the proxy use differential privacy and add noise to protect individuals from the possibility of targeting.  I stopped short of that though on the basis that we don't really need to litigate the details here: knowledge of the nature of the information that is added should be enough.

Another thing that might arise is the potential for optional decoration being omitted.  Let's say that you have a Spam-Score[^1] flag that carries a limited number of discrete values (integers 0 to 10, say).  Users are aware that they are being rated, so that is broadly OK.  If the proxy doesn't attach any score to a request, does that signify something that singles out that client?  Again, I think that this is implicit, but I worry that it isn't.

[^1]: Your perspective on this might look very different if you call this Social-Score...

Reply:
> Another thing that might arise is the potential for optional decoration being omitted. Let's say that you have a Spam-Score1 flag that carries a limited number of discrete values (integers 0 to 10, say). Users are aware that they are being rated, so that is broadly OK. If the proxy doesn't attach any score to a request, does that signify something that singles out that client? Again, I think that this is implicit, but I worry that it isn't.

Indeed, this is somewhat related to #114, and is likely worth a larger discussion about the interaction of signals from proxy to target and vice versa.